lib/get-esm-exports.js

```diff
@@ -0,0 +1,93 @@
+'use strict'
+
+const { Parser } = require('acorn')
+const { importAssertions } = require('acorn-import-assertions');
+
+const acornOpts = {
+  ecmaVersion: 'latest',
+  sourceType: 'module'
+}
+
+const parser = Parser.extend(importAssertions)
+
+function getEsmExports (moduleStr) {
+  const exportedNames = new Set()
+  const tree = parser.parse(moduleStr, acornOpts)
+  for (const node of tree.body) {
+    if (!node.type.startsWith('Export')) continue
+    switch (node.type) {
+      case 'ExportNamedDeclaration':
+        if (node.declaration) {
+          parseDeclaration(node, exportedNames)
+        } else {
+          parseSpecifiers(node, exportedNames)
+        }
+        break
+      case 'ExportDefaultDeclaration':
+        exportedNames.add('default')
+        break
+      case 'ExportAllDeclaration':
+        if (node.exported) {
+          exportedNames.add(node.exported.name)
+        } else {
+          exportedNames.add('*')
+        }
+        break
+      default:
+        throw new Error('unrecognized export type: ' + node.type)
```

Comment:
Would throwing interrupt the loading entirely? Perhaps we should do console.error instead?

Reply:
Yeah as @tlhunter suggested, maybe that or `process.emitWarning()`